### PR TITLE
Kubernetes: lookup cluster config from google api when using GKE

### PR DESCRIPTION
**Pull request by mclarke47**
## Hey, I just made a Pull Request!

Add the ability to lookup cluster configuration from the GKE API, so the config setup is easier.

There were a few other boilerplate-y changes needed but it should be much, much easier to add new cluster locator methods now.

For now this cluster locator method will assume use of the `google` auth provider, but this can be changed later to use a gcloud service account or possibly even a kubernetes service account later, to make it more flexible.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation 
- [x] Tests for new functionality and regression tests for bug fixes

### CODE REVIEW

**Comment by mclarke47** [comment]
The config will be backwards incompatible. Which is why it's a major changeset bump @Rugvip lemme know if it should be a minor instead,

File: .changeset/shaggy-islands-mix.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kubernetes': major
+'@backstage/plugin-kubernetes-backend': major
```
Comment (Rugvip) [inline .changeset/shaggy-islands-mix.md:3]: It should totally be a minor instead, we're sticking to 0.x for now :grin:

File: .changeset/shaggy-islands-mix.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Restructure configuration; Add GKE cluster locator
```
Comment (Rugvip) [inline .changeset/shaggy-islands-mix.md:6]: Not required, but always good if you can give a few pointers how to migrate at least at a high level

**Review by Rugvip — COMMENTED** [review]
Just had a look at the changeset for now

**Comment by mclarke47** [comment]
It's ready for review, just need to do some testing. Probably tomorrow(ish).

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Comment by mclarke47** [comment]
Tested it out and it works as expected! 🎉

**Comment by benjdlambert** [comment]
Hey @mclarke47 - you can rebase this on top of master now and it will fix the failing test 👍

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4844-uqfifojwfwnm0ccq3romoktpjxooen3f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60428a81f285af4e1da1a8cb

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**